lib/active_shipping/carriers/ups.rb

```diff
@@ -450,15 +450,23 @@ def build_shipment_request(origin, destination, packages, options={})
             end
           end
 
-          # I don't know all of the options that UPS supports for labels
-          # so I'm going with something very simple for now.
+          # defaults to GIF
+          image_format = options[:image_format].nil? ? 'GIF' : options[:image_format].upcase
+
           xml.LabelSpecification do
+            # Support ZPL Files
+            if image_format == 'ZPL'
+              xml.LabelStockSize do
+                xml.Height('4')
+                xml.Width('6')
+              end
+            end
             xml.LabelPrintMethod do
-              xml.Code('GIF')
+               xml.Code(image_format)
             end
             xml.HTTPUserAgent('Mozilla/4.5') # hmmm
```

Comment:
According to the UPS docs this HTTPUserAgent node is actually only required if the LabelPrintMethod == 'GIF', and the LabelStockSize above actually applies to all of the non GIF formats (EPL, ZPL, STARPL and SPL), so would you mind flipping this conditional around so that the HTTPUserAgent is included if the LabelPrintMethod is GIF and the LabelStockSize is included in all other cases?